src/app/[lang]/admin/login/page.tsx

```diff
@@ -0,0 +1,69 @@
+"use client";
+
+import { Formik } from "formik";
+import * as yup from "yup";
+
+let LoginSchema = yup.object().shape({
+  username: yup.string().required("Please enter your username"),
+  password: yup
+    .string()
+    .required("Password is a required field")
+    .min(8, "Password must be at least 8 characters"),
+});
+
+export default function Login() {
+  return (
+    <div className="min-h-screen flex items-center">
+      <div className="text-center w-full">
+        <h2>Welcome to admin panel</h2>
+        <div className="mt-10">
+          <Formik
+            validationSchema={LoginSchema}
+            initialValues={{ username: "", password: "" }}
+            onSubmit={(values) => {
+              alert(JSON.stringify(values));
+            }}
+          >
+            {({
+              values,
+              errors,
+              touched,
+              handleChange,
+              handleBlur,
+              handleSubmit,
+            }) => (
+              <div className="login">
+                <div className="form">
+                  <form noValidate onSubmit={handleSubmit}>
+                    <input
+                      type="text"
+                      name="username"
+                      onChange={handleChange}
+                      value={values.username}
+                      placeholder="Enter username"
+                    />
+                    <p className="text-center text-red-900">
+                      {errors.username && touched.username && errors.username}
+                    </p>
+                    <input
+                      type="password"
+                      name="password"
+                      onChange={handleChange}
+                      onBlur={handleBlur}
+                      value={values.password}
+                      placeholder="Enter password"
+                    />
+                    <p className="text-center text-red-900">
+                      {errors.password && touched.password && errors.password}
```

Comment:
Also, I'd suggest you to create a custom component for `Input`

```tsx

type Props = React.InputHTMLAttributes<HTMLInputElement> & {
  label?: React.ReactNode;
  error?: React.ReactNode;
};

export const Input = (props: Props) => {
  const { label, error, ...restProps } = props;

  return (
    <div className="flex flex-col">
      {label && <label>{label}</label>}
      <input {...restProps} />
      {error && <p className="text-red-900">{error}</p>}
    </div>
  );
};
```

So then you can use it as regular input, but also pass `label` and `error` props.

```tsx
<Input
  value={someValue}
  onChange={console.log}
  label="This is input"
  error="Must be equal 0"
/>
```